### PR TITLE
Implement actions using userOpBuilder service instead of permissionless.js

### PR DESCRIPTION
**Pull request by KannuSingh**
# Description
- refactored permissions implementations
- using UserOpBuilder service instead of UserOpBuilder Contract for building userOperation.
- removed `permissionless.js` dependency

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections

### CODE REVIEW

File: apps/laboratory/src/utils/WalletConnectCosignerUtils.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  }
+}
+
+// Class to interact with the WalletConnect CoSigner API
+export class WalletConnectCosigner {
+  private baseUrl: string
+  private projectId: string
+
+  constructor() {
+    this.baseUrl = 'https://rpc.walletconnect.com/v1/sessions'
```
Comment (lukaisailovic) [inline apps/laboratory/src/utils/WalletConnectCosignerUtils.ts:113]: Should be const

File: apps/laboratory/src/utils/UserOpBuilderServiceUtils.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    baseUrl,
+    transformedArgs
+  )
+
+  if ('error' in response.data) {
+    throw new Error(response.data.error)
+  }
+
+  return response.data
+}
```
Comment (lukaisailovic) [inline apps/laboratory/src/utils/UserOpBuilderServiceUtils.ts:84]: Duplicated code, also url is in code it should be in the const

File: apps/laboratory/src/hooks/useERC7715Permissions.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        expiry: approvedPermissions.expiry,
+        signer: {
+          type: 'donut-purchase',
+          data: {
+            ids: [addPermissionResponse.key, hexStringToBase64(signerKey.key)]
+          }
+        },
+        signerData: {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
+          userOpBuilder: approvedPermissions.signerData?.userOpBuilder!
```
Comment (lukaisailovic) [inline apps/laboratory/src/hooks/useERC7715Permissions.ts:91]: Why this? do we need the type change from viem?

Comment (KannuSingh) [reply apps/laboratory/src/hooks/useERC7715Permissions.ts:91]: @lukaisailovic wdym, do we need to pass this value to cosigner or why are we using viem type? or both
- current change cosigner don't need `userOpBuilder` so can be set to empty string.
- walletClient.grantPermissions returned `GrantPermissionsReturnType` viem type for approvedPermissions

**Review by lukaisailovic — APPROVED** [review]

**Pull request merged**